### PR TITLE
Adds Norf guns

### PR DESCRIPTION
**Pull request by DrSnips**
<h1>WIP</h1>
Adds in norf guns that fire foam darts. They come in many types like handguns, SMGs, gat, rifles, etc.
Most of them use a mag for their ammo but there are some exceptions like the handguns. This PR is heavy WIP right now and is mostly framework for all future norf guns. There are some issues here and there like how the darts function. Currently the darts "stick" to where they hit,this is a FEATURE, including walls this leads to some of the rapid fire ones hitting yourself if you run in the direction you're shooting.
To do list:

- [x] Add in the basic framework.

- [ ] Make sure all of the basic framework works.

- [x] Add handgun base. (no mag)

- [x] Add SMG/AR base. (mag)

- [ ] Add heavy guns base.

- [ ] Add shotgun base.

- [ ] Add rifle base.

- [ ] Add a way to get the guns. (open to suggestions but I'm currently thinking a vending machine)

Possible to do list:
- [ ] Add different colored ammo.

- [ ] Add some attachments for the tactical rails.

- [ ] Add more than one gun of each type.

- [ ] Mech mounted blaster.

:cl:
 * rscadd: Added norf guns. They come in a variety of types similar to normal guns (handgun, SMG, gat, rifle, etc.)

### CODE REVIEW

**Comment by DrSnips** [comment]
This PR is going to require a lot of sprite work. The only things done right now are the ammo/mags. I have a list of planned guns to start with if you want to pitch in but can't decide which one to sprite.
Rifle: longshot(+detaching HG), longstrike
MG: vulcan
AR: raider, recon
Shotgun: rough cut
SMG: deploy
Handgun: maverick, stryfe

**Comment by Dacendeth** [comment]
While a vending machine with them would be nice, some if not all should be added to the arcade machine rewards as well.

**Comment by Kitfox88** [comment]
Do you mean you want the sprites to resemble those IRL nerf guns, snips? For the list of shit, that is.

**Comment by DrSnips** [comment]
Yeah that would be the ideal first wave of them where they are similar to the source. I'm not turning away other guns I just polled people on the discord for a basic line up of nerfguns they would like to see to start it.

**Comment by Chazach** [comment]
the pr that saved /vg/

**Comment by AI-Is-Rogue** [comment]
Norf mech when?

**Comment by DrSnips** [comment]
I can add a mech dart blaster to the possible to dos.

**Comment by spoonylove** [comment]
Are you going to port TGs lethal foam darts?

**Comment by DrSnips** [comment]
I'd rather not. I'd like them to be as non-controversial as possible.

**Comment by Kitfox88** [comment]
maverick
Is this sort of thing okay, or should I get more practice before trying this shit?

**Comment by Durgen** [comment]
can clumsy use these?

File: code/modules/projectiles/guns/projectile/foam.dm [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	w_class = W_CLASS_SMALL
+	origin_tech = null //Tc_COMBAT + "=4;" + Tc_MATERIALS + "=2"
+	fire_sound = 'sound/items/syringeproj.ogg'
+	max_shells = 1
+	caliber = list(FOAM = 1)
+	ammo_type ="/obj/item/ammo_casing/foam"
+	load_method = 0
+	recoil = 0
+	gun_flags = 0
+	clumsy_check = FALSE //So nerf guns don't explode on clowns.
```
Comment (DrSnips) [inline code/modules/projectiles/guns/projectile/foam.dm:17]: @Durgen yes.

**Comment by DrSnips** [comment]
@Kitfox88 Looks pretty good to me. Mind if I add it in? You can always just send another if you update it.

**Comment by Inorien** [comment]
Yo I'll sprite you some, i love these
I say put them in a cargo order (either directly or in vending charge pack form)

**Comment by DrSnips** [comment]
I swear this PR isn't dead. Because of the conflicts on the byond version I don't want to push stuff to here so I will be making a new PR for all of this when it's done. Expect Norf Guns II: Attack of the Foams soonTM.

**Pull request closed without merging**